### PR TITLE
Ignore hidden folders for image_dataset_from_directory

### PR DESCRIPTION
**Pull request by sachinprasadhs**
Ignore hidden folders for image_dataset_from_directory like .git .ipynb etc.
Fixes: https://github.com/keras-team/keras/issues/18170

### CODE REVIEW

**Comment by ARPIT2128** [comment]
Lint will reject it due to `IndentationError` try in one line 😃

File: keras/utils/dataset_utils.py [hunk 1]
```diff
@@ -540,7 +540,10 @@ def index_directory(
     else:
         subdirs = []
         for subdir in sorted(tf.io.gfile.listdir(directory)):
-            if tf.io.gfile.isdir(tf.io.gfile.join(directory, subdir)):
+            if (
```
[the remark this answers is not in the document]
Comment (sachinprasadhs) [reply keras/utils/dataset_utils.py]: Still it gives lint error since it exceeds 80 character limit

**Comment by nkovela1** [comment]
Please fix the code formatting. The change LGTM otherwise!

File: keras/utils/dataset_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ARPIT2128) [reply keras/utils/dataset_utils.py]: please try


```
if (tf.io.gfile.isdir(tf.io.gfile.join(directory, subdir)):
    if(subdir.startswith(".")):
      <code>
    else:
     <code>
```

[the remark this answers is not in the document]
Comment (sachinprasadhs) [reply keras/utils/dataset_utils.py]: I always wanted to avoid adding too many. if conditions, I guess I will have to do this

**Comment by ARPIT2128** [comment]
https://github.com/keras-team/keras/pull/18177#discussion_r1213959676

**Comment by sachinprasadhs** [comment]
@nkovela1 Could you please review it.

**Review by nkovela1 — APPROVED** [review]
Adding @fchollet for merge access. LGTM!

**Pull request merged**